starknet_devnet/cairo_rs_py_patch.py

```diff
@@ -0,0 +1,577 @@
+    initial_os_context: List[MaybeRelocatable],
+):
+    """
+    Validates and processes an OS context that was returned by a transaction.
+    Returns the syscall processor object containing the accumulated syscall information.
+    """
+    os_context_end = runner.get_ap() - 2
+    stack_ptr = os_context_end
+    # The returned values are os_context, retdata_size, retdata_ptr.
+    stack_ptr = runner.get_builtins_final_stack(stack_ptr)
+
+    final_os_context_ptr = stack_ptr - 1
+    assert final_os_context_ptr + len(initial_os_context) == os_context_end
+
+    # Validate system calls.
+    syscall_base_ptr, syscall_stop_ptr = segment_utils.get_os_segment_ptr_range(
+        runner=runner, ptr_offset=SYSCALL_PTR_OFFSET, os_context=initial_os_context
+    )
+
+    segment_utils.validate_segment_pointers(
+        segments=runner,
+        segment_base_ptr=syscall_base_ptr,
+        segment_stop_ptr=syscall_stop_ptr,
+    )
+    syscall_handler.post_run(runner=runner, syscall_stop_ptr=syscall_stop_ptr)
+
+
+def cairo_rs_py_allocate_segment(
+    self, segments: MemorySegmentManager, data: Iterable[MaybeRelocatable]
+) -> RelocatableValue:
+    try:
+        segment_start = segments.add_segment()
+    except:
+        segment_start = segments.add()
+
+    segment_end = segments.write_arg(ptr=segment_start, arg=data)
+    self.read_only_segments.append((segment_start, segment_end - segment_start))
+    return segment_start
+
+
+def cairo_rs_py_read_and_validate_syscall_request(
+    self,
+    syscall_name: str,
+    segments: MemorySegmentManager,
+    syscall_ptr: RelocatableValue,
+) -> CairoStructProxy:
+    """
+    Returns the system call request written in the syscall segment, starting at syscall_ptr.
+    Performs validations on the request.
+    """
+    # Update syscall count.
+    self._count_syscall(syscall_name=syscall_name)
+
+    request = self._read_syscall_request(
+        syscall_name=syscall_name, segments=segments, syscall_ptr=syscall_ptr
+    )
+
+    assert (
+        syscall_ptr == self.expected_syscall_ptr
+    ), f"Bad syscall_ptr, Expected {self.expected_syscall_ptr}, got {syscall_ptr}."
+
+    syscall_info = self.syscall_info[syscall_name]
+    self.expected_syscall_ptr += syscall_info.syscall_size
+
+    selector = request.selector
+    assert isinstance(selector, int), (
+        f"The selector argument to syscall {syscall_name} is of unexpected type. "
+        f"Expected: int; got: {type(selector).__name__}."
+    )
+    assert (
+        selector == syscall_info.selector
+    ), f"Bad syscall selector, expected {syscall_info.selector}. Got: {selector}"
+
+    args_struct_def: StructDefinition = (
+        syscall_info.syscall_request_struct.struct_definition_
+    )
+    for arg, (arg_name, arg_def) in safe_zip(request, args_struct_def.members.items()):
+        expected_type = get_runtime_type(arg_def.cairo_type)
+    #     assert isinstance(arg, expected_type), (
```

Comment:
I see there are lines like this in several places that are still commented. Why is this so? Is there a better approach?